src/components/Nav.vue

```diff
@@ -1,16 +1,156 @@
 <template>
-  <header>
-    <div><img src="/logo.png" alt="rick and morty" /></div>
-    <nav>
-      <RouterLink v-for="{ route, label } in navRoutes" :to="route">
-        {{ label }}
+  <header class="nav">
+    <div class="nav__logo-container">
+      <RouterLink to="/">
+        <img 
+          class="nav__logo" 
+          src="/logo.png" 
+          alt="rick and morty" 
+        />
       </RouterLink>
+    </div>
+    <nav class="nav__nav-container">
+      <ul ref="navList" class="nav__nav-list">
+        <li 
+          v-for="{ route, label } in navRoutes" 
+          ref="routeContainer" 
+          class="nav__nav-list-item"
+        >
+          <RouterLink 
+            class="nav__link" 
+            :to="route" 
+            active-class="active"
+          >
+            {{ label }}
+          </RouterLink>
+        </li>
+      </ul>
     </nav>
   </header>
 </template>
 
 <script setup lang="ts">
-import { RouterLink } from 'vue-router'
+import { RouterLink, useRoute } from 'vue-router'
 
 import { navRoutes } from '@/constants'
+import { nextTick, useTemplateRef, watch } from 'vue'
+
+const navList = useTemplateRef<HTMLUListElement | null>('navList')
+const routesContainers = useTemplateRef<HTMLAnchorElement[] | null>('routeContainer')
+
+const route = useRoute()
+
+const findActiveRouteElement = () =>
```

Comment:
in vue 3 it makes sense to use regular functions instead of arrow for methods.
It improves readability a bit and we can use hoisting to keep methods on the bottom and access them from top lines.